### ISSUE TITLE
Why would you use this library?

### ISSUE THREAD

**Issue by maandree** [clibs/int#3]
First of: ”License: MIT”, is not a sufficient way to specify the license.
There are two licenses known as ”the MIT License”, but none if them
is named ”MIT License”. Please a full LICENSE file (this time with
the XML entities.)

But to the issue.

You seldom need big endian and little endian (or mixed endian) read and
write functions for integer types. Most often, what you need is host endian
and network endian. For host endian no function is necessary, but if you
want something:

```
#define READ_HOST_INT(buf, type)  (*(type*)buf)
#define WRITE_HOST_INT(buf, n)  memcpy(buf, &(n), sizeof(n))
```

These have to advantage to work on all integer types.
For network endian:

```
#define READ_NETWORK_INT16(buf)  ntohs(READ_HOST_INT(buf, uint16_t))
#define READ_NETWORK_INT32(buf)  ntohl(READ_HOST_INT(buf, uint32_t))
#define WRITE_NETWORK_INT16(buf, n)  WRITE_HOST_INT(buf, htons(n))
#define WRITE_NETWORK_INT32(buf, n)  WRITE_HOST_INT(buf, htonl(n))
```

There is not standard functions for 64bits here, unfortunely. There is ofcourse
no need for other types than exactly 16bits, 32bits and 64bits, because these
and the trivial 8bits, are the only ones that are portable. (not `{,u}int[0-9]*_t` types are portable.)
For network endian, signed integers, values most be converted to two's complement.

To sum up, what we need is 64-bit conversion between host and endian, and
convesion between two's complement and what ever processes uses (not
necessarily what the machine uses natively). That latter is less important,
because no sensible C compiler for POSIX systems would use anything else
than two's complement.

**Comment by tj** [comment]
Don't use it? problem solved

**Comment by maandree** [comment]
It was suggested to me that I should file a ticket to let you know. This is simply for your edification.

**Comment by tj** [comment]
I definitely have a bad habit of not including the full license. My apologies!   Critique is always welcome of course, it just seemed like you had no intention of using it anyway.
